docs/tutorial.md

```diff
@@ -0,0 +1,586 @@
+# Charmed MySQL K8s tutorial
+The Charmed MySQL K8s Operator delivers automated operations management from [day 0 to day 2](https://codilime.com/blog/day-0-day-1-day-2-the-software-lifecycle-in-the-cloud-age/) on the [MySQL Community Edition](https://www.mysql.com/products/community/) relational database. It is an open source, end-to-end, production-ready data platform [on top of Juju](https://juju.is/). As a first step this tutorial shows you how to get Charmed MySQL K8s up and running, but the tutorial does not stop there. Through this tutorial you will learn a variety of operations, everything from adding replicas to advanced operations such as enabling Transport Layer Security (TLS). In this tutorial we will walk through how to:
+- Set up your environment using [Microk8s](https://microk8s.io/) and Juju.
+- Deploy MySQL using a single command.
+- Access the admin database directly.
+- Add high availability with MySQL InnoDB Cluster, Group Replication.
+- Request and change the admin password.
+- Automatically create MySQL users via Juju relations.
+- Reconfigure TLS certificate in one command.
+
+While this tutorial intends to guide and teach you as you deploy Charmed MySQL K8s, it will be most beneficial if you already have a familiarity with:
+- Basic terminal commands.
+- MySQL concepts such as replication and users.
+
+## Minimum requirements
+Before we start, make sure your machine meets the following requirements:
+- Ubuntu 20.04 (Focal) or later.
+- 8GB of RAM.
+- 2 CPU threads.
+- At least 20GB of available storage.
+- Access to the internet for downloading the required snaps and charms.
+
+
+## Prepare Microk8s
+The fastest, simplest way to get started with Charmed MySQL K8s is to set up a local Microk8s lightweight Kubernetes. Microk8s is the best Kubernetes experience for developers, DevOps, cloud and edge; Charmed MySQL K8s will be run in Microk8s Kubernetes and managed by Juju. While this tutorial covers the basics of Microk8s, you can [explore more Microk8s here](https://microk8s.io/). To install Microk8s on Ubuntu 22.04 execute:
```

Comment:
One recommendation if I may -- I find starting the tutorial with getting Multipass installed and running the tutorial in the fresh VM is a nice way of counteracting any weirdness on the reader's system :)

Reply:
(It also means at the end of the tutorial they only ever have to `multipass delete <name>` to clean up!)

Reply:
Tnx @jnsgruk for the feedback!
I wish to use multipass (in fact I am using it locally already, but constantly facing [this issue](https://github.com/canonical/multipass/issues/2772).
Maybe you can rise a priority there due to high visibility in the first copy&paste command?

Reply:
Done in https://github.com/canonical/mysql-k8s-operator/pull/148/commits/088207a84cd5ae45fbc7c6db25d8f309503855f1. Resolved.